### PR TITLE
Jakarta EE 9 Platform Specification

### PR DESCRIPTION
**Pull request by kwsutter**
Signed-off-by: Kevin Sutter <kwsutter@gmail.com>

## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [x] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [x] N/A - For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
- [x] For a [Release Review](https://www.eclipse.org/projects/handbook/#release-review), a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification. The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
   - [x] Updated release record
   - [x] Generated IP Log
   - [x] Email to PMC
   - [x] Start release review by emailing EMO
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/repositories/jakartaplatform-1013/jakarta/platform/jakarta.jakartaee-api/9.0.0/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted/jakarta-jakartaeetck-9.0.0.zip
- [x] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/jakartaee-platform/issues/258
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

Note: If any item does not apply, check it and mark N/A below it.

### CODE REVIEW

**Comment by kwsutter** [comment]
Creating this placeholder PR for the Platform Specification.  It's not ready for final review yet.  Scott Wang (Primeton) is the assigned mentor/reviewer for the Platform PR.  I will add the PR Checklist to kick this off in case somebody wants to help out Scott.  Thanks.

**Comment by kwsutter** [comment]
# Spec Review Checklist

1. Spec PR
  - [ ] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form {spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form {spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta.{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request
 
10. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

**Comment by ScottWangNL** [comment]
I found that these two file names seem to be wrong: 

- platform-spec-9-SNAPSHOT.html   should be platform-spec-9.html
- platform-spec-9-SNAPSHOT.pdf should be platform-spec-9.pdf

In specifications/platform/9/_index.md, the links point to these two files don't seem to work.

**Comment by ScottWangNL** [comment]
In specifications/platform/9/_index.md file, the link point to "Jakarta EE Platform 9 TCK" doesn't work too.

**Comment by ScottWangNL** [comment]
In specifications/platform/_index.md file, there is an extra column called "weight", which I haven't found in other specs, I am not sure if it is reasonable.

File: platform/9/_index.md [hunk 1]
```diff
@@ -1,24 +1,24 @@
 ---
-title: "Jakarta EE Platform 9 (milestone)"
-date: 2020-06-15
+title: "Jakarta EE Platform 9"
+date: 2020-09-01
```
Comment (ScottWangNL) [inline platform/9/_index.md:3]: as Kevin's mail mentioned, this date should be move to Nov 1 ?

**Comment by kwsutter** [comment]
> In specifications/platform/_index.md file, there is an extra column called "weight", which I haven't found in other specs, I am not sure if it is reasonable.

Very observant!  :-) This weight field is used to help order the items on the jakarta.ee specifications page.  We want the Platform first, then Web Profile, then alphabetical (default).  So, that's why you will find a weight in the Platform and Web Profile _index.md files, but none of the others.

**Comment by kwsutter** [comment]
The Spec, API, and Javadoc are ready for review.  The javadoc is missing the EFSL -- working on that.  Enjoy!

**Comment by hussainnm** [comment]
> The Spec, API, and Javadoc are ready for review. The javadoc is missing the EFSL -- working on that. Enjoy!

Fixed the javadoc issue in https://github.com/eclipse-ee4j/jakartaee-api/pull/90

**Comment by Emily-Jiang** [comment]
@kwsutter it seems I am unable to click the checkbox. Does this mean only comitters on this project can edit the checklist?

**Comment by kwsutter** [comment]
> @kwsutter it seems I am unable to click the checkbox. Does this mean only comitters on this project can edit the checklist?

Correct, @Emily-Jiang.  The checklist is supposed to be used by the Spec Committee to verify that this PR is ready for ballot.  If you find something that needs correcting, please just post a comment to get it corrected.  But, the checkboxes themselves need to be marked off by a committer on this project.  Thanks!

**Comment by kwsutter** [comment]
@jeanouii Just noticed that you are listed as a mentor for the Platform.  This is pretty much ready to go.  I'm waiting on EJB to get to ballot before recreating the APIs and Javadoc (with the license update).  Please review away so that we can have this ready by Friday. Thanks!

**Review by jeanouii — APPROVED** [review]
I'll put the ballot up now. 
Please don't update the PR from now on ;-)

**Comment by kwsutter** [comment]
Per the Spec Committee meeting on 11/11, I updated the reference to the TCK to be without the "-RC1".  Since the same SHA values exist for both versions, this was deemed an acceptable and expected update.

**Comment by jeanouii** [comment]
- [ ] adds this final checklist to the main PR.
 - [ ] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [ ] creates an issue in the specification project that includes the following checklist for the specification project team:
https://github.com/eclipse-ee4j/authentication/issues/114

**Pull request merged**